respx/api.py

```diff
@@ -1,8 +1,10 @@
-from typing import Callable, Optional, Pattern, Union
+from typing import Any, Callable, Optional, Pattern, TypeVar, Union, overload
 
 from .mocks import MockTransport
 from .models import ContentDataTypes, HeaderTypes, RequestPattern
 
+DefaultType = TypeVar("DefaultType", bound=Any)
+
```

Comment:
Just a minor, maybe this `DefaultType` could be moved to `models.py`, to be used both in api and transports modules, instead of duplicating?